### ISSUE TITLE
inputRef or ref prop doesn't work

### ISSUE THREAD

**Issue by sadik-malik** [dohomi/react-hook-form-mui#214]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

In @mui/material, we can pass the ref prop to the TextField component using the inputRef prop. However, with TextFieldElement, we cannot do the same.

If we want to focus on the TextField component programmatically for some reason at some point in time, then we cannot do the same. This issue exists for maybe all of the components in react-hook-form-mui.

### Expected behavior 🤔

If we are passing inputRef prop to TextFieldElement and want to focus input programmatically using inputRef.current.focus(), then it should work. 

### Steps to reproduce 🕹

I have created demo for it in this [sandbox](https://codesandbox.io/s/elated-bell-txwlq4) along with its fix using useController and useForkRef. I believe this fix can be extended to other components as well.

**Comment by dohomi** [comment]
@sadik-malik this is a great find, I didn't know about `userForkRef` helper hook. Basically it makes sense to adjust all components with that feature. I might think probably its not a bad idea to move most components from `<Controller />` to a `useController` hook and move common logic to a custom hook which takes care of `useController` and `useForkRef` and maybe some other common functions

**Comment by sadik-malik** [comment]
Sure, I will be happy to make the changes.

**Comment by sadik-malik** [comment]
I have made changes in my repository, but these changes include quite a lot of modifications. Can we create a separate branch for this, where we can merge the code and write the Storybook examples and test the code?

**Comment by dohomi** [comment]
@sadik-malik I published a branch v7
<img width="458" alt="Screenshot 2023-11-21 at 14 16 00" src="https://github.com/dohomi/react-hook-form-mui/assets/489221/a3e52713-bbc2-4d78-891f-19080260926e">

**Comment by sadik-malik** [comment]
@dohomi raised the [PR](https://github.com/dohomi/react-hook-form-mui/pull/215) for v7 branch.

**Comment by dohomi** [comment]
@sadik-malik thanks for your great contribution I took the v7 branch and merged it into master. I found a few breaking changes which I fixed along the line and I also added Chromatic as a testrunner to safely test now all component stories.

I had some issues with the `lint-staged` script as it was running falsely over `storybook-static` and throw hundreds of errors along. I removed if now but feel free to re-add it but make sure you ran storybook build before. I think .eslintignore needs to be add but somehow the file is ignored
